media/js/mozorg/contribute/taskview.js

```diff
@@ -0,0 +1,316 @@
+$(function() {
+    'use strict';
+
+    var $taskSteps = $('.task-steps');
+    var $stepOne = $('#step_one');
+    var $thankYou = $('#thankyou');
+    var $getFirefox = $('.get-firefox');
+    var $actionButton = $('.action');
+    var $downloadButton = $('.download-link');
+
+    var $followButton = $('.follow-mozilla');
+    var intentURL = $followButton.attr('href');
+
+    var $signupTweetForm = $('#signup-tweet');
+    var $downloadDevTools = $('.dev-edition');
+    var $tryAnotherTask = $('.try-another');
+
+    // some tasks, like installing Whimsy, required the user to be using Firefox
+    if ($getFirefox.length > -1 && !Mozilla.Client.isFirefox) {
+        $getFirefox.toggleClass('hidden');
+        // hide any action buttons that forms part of the task.
+        $actionButton.addClass('hidden');
+    }
+
+    /**
+     * Determines the visibilitychange event name based on the current browser
+     * and returns the appropriate keyword. Based on code from:
+     * https://developer.mozilla.org/en-US/docs/Web/API/Page_Visibility_API
+     */
+    function getVisibilityStateEventKeyword() {
+        // for browser that support the event unprefixed
+        if (typeof document.hidden !== 'undefined') {
+            return 'visibilitychange'
+        }
+
+        // for Chrome 13 and lower
+        if (typeof document.webkitHidden !== 'undefined') {
+            return 'webkitvisibilitychange'
+        }
+
+        // for older versions of IE
+        if (typeof document.msHidden !== 'undefined') {
+            return 'msvisibilitychange'
+        }
+    }
+
+    /**
+     * Toggles the completed class on the relevant step, and calls
+     * taskComplete() if this was the final step of the task.
+     */
+    function completeStep($step) {
+        var $stepOne = $('#step_one');
+        var $stepTwo = $('#step_two');
+
+        if ($step.data('step') === 'one') {
+            $stepOne.toggleClass('completed');
+        } else if ($step.data('step') === 'two') {
+            $stepTwo.toggleClass('completed');
+        }
+
+        if ($step.data('complete') === true) {
+            taskComplete();
+        }
+    }
+
+    /**
+     * Called once all steps of the task has been completed. This will
+     * show the thank message and scroll it into view.
+     */
+    function taskComplete() {
+        $thankYou.removeClass('visibly-hidden');
+        $thankYou.attr('aria-hidden', 'false');
+        $thankYou[0].scrollIntoView();
+        $thankYou.focus();
+    }
+
+    /**
+     * Waits for the initial tab/window to become visible, and then
+     * proceeds to complete the relevant task step.
+     */
+    function handleVisibilityChange($step) {
+
+        var $document = $(document);
+        var visibilityChange = getVisibilityStateEventKeyword();
```

Comment:
Since this function *can* be called twice on one page (Whimsy V2), I'd suggest moving `$document` and `visibilityChange` up a level in scope so they doesn't get re-declared/initialized on each call to `handleVisibilityChange`.